main.go

```diff
@@ -88,9 +88,24 @@ func main() {
 		exit(1)
 	}
 
-	adds, deletes := localRecords.Diff(records)
+	// Cast the records from Cloudflare to a recordCollection.
+	existingRecords := recordCollection(records)
 
-	numChanges := len(adds) + len(deletes)
+	// Find records only present at cloudflare - and records only present in
+	// the local zone. This will be the basis for the add/delete collections.
+	addCandidates := fileRecords.Difference(existingRecords, FullMatch)
+	deleteCandidates := existingRecords.Difference(fileRecords, FullMatch)
+
+	// If we find the intersection between local and remote, we should have a
+	// list of records to update. We use only Updatable here, because that
+	// will give us a collection of records that makes sense to update.
+	updates := deleteCandidates.Intersect(addCandidates, Updatable)
+
+	// The changed records can be removed from the add and delete slices.
```

Comment:
No records have actually been changed yet. How about "The records up for updating..."?